src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier colorsupplier = new ColorSupplier();
```

Comment:
Think about which variables should be local in the method and which should be class-level. The 'ColorSupplier' instance should not be a class-level variable in 'RightTriangle'. Instead, it should be used locally where needed or passed as a parameter.